packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -105,7 +271,40 @@ function ListViewBlockSelectButton(
 						</span>
 					) }
 				</HStack>
-			</Button>
+
+				<BlockSettingsMenuControls>
```

Comment:
I don't think this should be part of the `Button` component. I know it's probably convenient to do so, but I think it makes the component confusing.

What was previously quite a simple component is now suddenly responsible for a lot, so it'd be nice to find a way to restructure this a bit.

Reply:
Let me take another look at how other items such as Locking work and see if it can be avoided.

Reply:
> What was previously quite a simple component is now suddenly responsible for a lot, so it'd be nice to find a way to restructure this a bit.

Agreed. I've taken on a bit of a refactor to try and separate out the behaviours a bit. At this point we have two components:

- `<ListViewBlockSelectButton>` - the component as it was in `trunk` but with some unavoidable additions to facilitate _triggering_ the renaming behaviour. Note that the actual renaming handling is now handled outside of this component.
- `<ListViewBlockInput>` - handles all the renaming behaviour and interactions.

This is different to what you and I discussed because I wanted to avoid the  `ListViewBlockSelectButton` being overcomplicated with concerns about managing the editing experience.

This has resulted in some duplication which we could look to refactor away in a follow up as required.